pkg/plugins/executable.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"sigs.k8s.io/kustomize/pkg/ifc"
+	"sigs.k8s.io/kustomize/pkg/pgmconfig"
+	"sigs.k8s.io/kustomize/pkg/resmap"
+)
+
+type ExecPlugin struct {
+	name string
+	arg  string
+	file string
+	rf   *resmap.Factory
+	ldr  ifc.Loader
+}
+
+func (p *ExecPlugin) Config(
```

Comment:
Please document the behavior of the two plugin fields `arg` and `file` as a comment on `Config`.

I think i'd prefer more self-explanatory field names
```
argOneLiner:  "admin secret"
argsFromFile: some/path/to/file.txt
```

Document that `file` (or `argsFromFile`) splits on new lines, and would be a way to pass, say, escaped spaces as arguments to the plugin.